### PR TITLE
chore: update how testing-library is used

### PR DESCRIPTION
**Pull request by dgdavid**
## TL;DR

This PR updates dependencies and changes how the testing-library is used.

## Motivation

There are two recommendations on how to use testing-library that I'd like to bring to the project:

* To prefer `screen` over _render queries_
* To use user-events instead of fireEvent

## Proposed changes

Adapt tests to follow these suggestions **after updating outdated dependencies**.

### Updated dependencies

Before running `npm update`  and a few `npm install <package-name>@latest`, the `npm outdated` warned about 19 packages.

<details>
<summary>Click to show/hide the `npm outdated` output</summary>

---

```
➜ npm outdated                        
Package                    Current   Wanted   Latest  Location                                Depended by
@babel/core                 7.18.5   7.21.0   7.21.0  node_modules/@babel/core                react-teleporter
@babel/eslint-parser        7.18.2   7.19.1   7.19.1  node_modules/@babel/eslint-parser       react-teleporter
@babel/preset-env           7.18.2   7.20.2   7.20.2  node_modules/@babel/preset-env          react-teleporter
@babel/preset-react        7.17.12   7.18.6   7.18.6  node_modules/@babel/preset-react        react-teleporter
@babel/preset-typescript   7.17.12   7.21.0   7.21.0  node_modules/@babel/preset-typescript   react-teleporter
@testing-library/jest-dom   5.16.4   5.16.5   5.16.5  node_modules/@testing-library/jest-dom  react-teleporter
@testing-library/react      13.3.0   13.4.0   14.0.0  node_modules/@testing-library/react     react-teleporter
@types/jest                 28.1.2   28.1.8   29.4.1  node_modules/@types/jest                react-teleporter
babel-jest                  28.1.1   28.1.3   29.5.0  node_modules/babel-jest                 react-teleporter
esbuild                    0.14.45  0.14.54  0.17.11  node_modules/esbuild                    react-teleporter
eslint                      8.18.0   8.36.0   8.36.0  node_modules/eslint                     react-teleporter
eslint-plugin-react         7.30.0   7.32.2   7.32.2  node_modules/eslint-plugin-react        react-teleporter
jest                        28.1.1   28.1.3   29.5.0  node_modules/jest                       react-teleporter
jest-environment-jsdom      28.1.1   28.1.3   29.5.0  node_modules/jest-environment-jsdom     react-teleporter
prettier                     2.7.1    2.8.4    2.8.4  node_modules/prettier                   react-teleporter
rollup                      2.75.6   2.79.1   3.19.1  node_modules/rollup                     react-teleporter
rollup-plugin-dts            4.2.2    4.2.3    5.2.0  node_modules/rollup-plugin-dts          react-teleporter
rollup-plugin-esbuild        4.9.1   4.10.3    5.0.0  node_modules/rollup-plugin-esbuild      react-teleporter
typescript                   4.7.4    4.9.5    4.9.5  node_modules/typescript                 react-teleporter

```
</details>

### Installed dependencies

[user-event](https://testing-library.com/docs/user-event/intro) @testing-library dependency has been added.

### Should the project stop using React.FC?

According to https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/function_components, it seems that `React.FC` is discouraged (?), reason why I was trying to get rid of it. However, I didn't manage to avoid complaints about `Component definition is missing display name` and finally discarded these changes.

### `npm audit` complaint

The  `npm audit` output says that there are `15 vulnerabilities (1 moderate, 14 high)`. It might be worthwhile to take a look at them.

<details>
<summary>Click to show/hide the `npm audit` output</summary>

---

```
# npm audit report

dot-prop  <4.2.1
Severity: high
dot-prop Prototype Pollution vulnerability - https://github.com/advisories/GHSA-ff7x-qrg7-qggm
fix available via `npm audit fix --force`
Will install conventional-github-releaser@1.1.7, which is a breaking change
node_modules/dot-prop
  compare-func  <=1.3.4
  Depends on vulnerable versions of dot-prop
  node_modules/compare-func
    conventional-changelog-angular  0.0.1 - 5.0.10
    Depends on vulnerable versions of compare-func
    node_modules/conventional-changelog-angular
      conventional-changelog  1.0.0 - 2.0.3
      Depends on vulnerable versions of conventional-changelog-angular
      node_modules/conventional-changelog
        conventional-github-releaser  >=1.1.0
        Depends on vulnerable versions of conventional-changelog
        Depends on vulnerable versions of gh-got
        Depends on vulnerable versions of git-semver-tags
        Depends on vulnerable versions of semver-regex
        node_modules/conventional-github-releaser

got  <=11.8.3
Severity: high
Got allows a redirect to a UNIX socket - https://github.com/advisories/GHSA-pfrx-2q88-qq97
Depends on vulnerable versions of cacheable-request
fix available via `npm audit fix --force`
Will install conventional-github-releaser@1.1.7, which is a breaking change
node_modules/got
  gh-got  <=9.0.0
  Depends on vulnerable versions of got
  node_modules/gh-got

http-cache-semantics  <4.1.1
Severity: high
http-cache-semantics vulnerable to Regular Expression Denial of Service - https://github.com/advisories/GHSA-rc47-6667-2j5j
fix available via `npm audit fix --force`
Will install conventional-github-releaser@1.1.7, which is a breaking change
node_modules/http-cache-semantics
  cacheable-request  0.1.0 - 2.1.4
  Depends on vulnerable versions of http-cache-semantics
  node_modules/cacheable-request

semver-regex  <=3.1.3
Severity: high
Regular expression denial of service in semver-regex - https://github.com/advisories/GHSA-4x5v-gmq8-25ch
Regular Expression Denial of Service (ReDOS) - https://github.com/advisories/GHSA-44c6-4v22-4mhx
fix available via `npm audit fix --force`
Will install conventional-github-releaser@1.1.7, which is a breaking change
node_modules/semver-regex

trim-newlines  <3.0.1
Severity: high
Uncontrolled Resource Consumption in trim-newlines - https://github.com/advisories/GHSA-7p7h-4mm5-852v
fix available via `npm audit fix --force`
Will install conventional-github-releaser@1.1.7, which is a breaking change
node_modules/get-pkg-repo/node_modules/trim-newlines
node_modules/git-raw-commits/node_modules/trim-newlines
node_modules/git-semver-tags/node_modules/trim-newlines
  meow  3.4.0 - 5.0.0
  Depends on vulnerable versions of trim-newlines
  node_modules/get-pkg-repo/node_modules/meow
  node_modules/git-raw-commits/node_modules/meow
  node_modules/git-semver-tags/node_modules/meow
    git-raw-commits  1.3.4 - 2.0.3
    Depends on vulnerable versions of meow
    node_modules/git-raw-commits
      conventional-changelog-core  2.0.5 - 4.2.1
      Depends on vulnerable versions of git-raw-commits
      Depends on vulnerable versions of git-semver-tags
      node_modules/conventional-changelog-core
    git-semver-tags  1.3.4 - 3.0.1
    Depends on vulnerable versions of meow
    node_modules/git-semver-tags

15 vulnerabilities (1 moderate, 14 high)

To address issues that do not require attention, run:
  npm audit fix

To address all issues (including breaking changes), run:
  npm audit fix --force

```

### CODE REVIEW

**Comment by gregberge** [comment]
@dgdavid maybe we could rebase this and upgrade the testing library?

**Comment by dgdavid** [comment]
> maybe we could rebase this and upgrade the testing library?

Sure! I'll do it during the day :)

**Comment by gregberge** [comment]
> Sure! I'll do it during the day :)

Thanks!

**Comment by dgdavid** [comment]
@gregberge done, the PR is now on top of `main` branch and the description has been updated. Please, have a look at least to the `npm audit` section. Thank you!

**Review by gregberge — APPROVED** [review]

**Comment by gregberge** [comment]
@dgdavid thanks!
